pkg/parse/mux.go

```diff
@@ -3,56 +3,21 @@ package parse
 import (
 	"net/http"
 
-	"github.com/gorilla/mux"
 	"github.com/rancher/apiserver/pkg/types"
 )
 
-type Vars struct {
-	Type      string
-	Name      string
-	Namespace string
-	Link      string
-	Prefix    string
-	Action    string
-}
-
-func Set(v Vars) mux.MatcherFunc {
-	return func(request *http.Request, match *mux.RouteMatch) bool {
-		if match.Vars == nil {
-			match.Vars = map[string]string{}
-		}
-		if v.Type != "" {
-			match.Vars["type"] = v.Type
-		}
-		if v.Name != "" {
-			match.Vars["name"] = v.Name
-		}
-		if v.Link != "" {
-			match.Vars["link"] = v.Link
-		}
-		if v.Prefix != "" {
-			match.Vars["prefix"] = v.Prefix
-		}
-		if v.Action != "" {
-			match.Vars["action"] = v.Action
-		}
-		if v.Namespace != "" {
-			match.Vars["namespace"] = v.Namespace
-		}
-		return true
-	}
-}
-
+// MuxURLParser is a URL parser that uses path variables from `net/http` routing.
+// This requires Go 1.22+ for path variable support in the standard library.
+// It is named MuxURLParser for backward compatibility, but no longer uses gorilla/mux.
 func MuxURLParser(rw http.ResponseWriter, req *http.Request, schemas *types.APISchemas) (ParsedURL, error) {
-	vars := mux.Vars(req)
 	url := ParsedURL{
-		Type:      vars["type"],
-		Name:      vars["name"],
-		Namespace: vars["namespace"],
-		Link:      vars["link"],
-		Prefix:    vars["prefix"],
+		Type:      req.PathValue("type"),
+		Name:      req.PathValue("name"),
+		Namespace: req.PathValue("namespace"),
+		Link:      req.PathValue("link"),
+		Prefix:    req.PathValue("prefix"),
 		Method:    req.Method,
-		Action:    vars["action"],
+		Action:    req.PathValue("action"),
```

Comment:
Can you verify if this still works for action and link?

From what I understand, gorilla/mux "vars" come from many places in the request like the path, query string, etc. I believe link and action would be ignored here since they're set in the query string, but we look for them in the URL path.

(any chance we can add some form of unit test here?)